include/boost/hana/experimental/printable.hpp

```diff
@@ -111,6 +111,18 @@ BOOST_HANA_NAMESPACE_BEGIN namespace experimental {
         }
     };
 
+    // model for hana::optional
+    template <>
+    struct print_impl<hana::optional_tag> {
+        template <typename O>
+        static auto apply(O const& optional) {
+            return hana::maybe("nothing",
+                [](auto x) {
+                    return "just " + hana::experimental::print(x);
```

Comment:
Instead of `"just x"`, I think it would be preferable to have `just(x)`. This way, if `x` is something complex, it might be easier to tell when the string representation of `x` finishes (because of the closing parenthesis). What do you think?